packages/toolkit/src/matchers.ts

```diff
@@ -404,22 +361,11 @@ export function isAsyncThunkAction<
     return isAsyncThunkAction()(asyncThunks[0])
   }
 
-  return (
-    action: any,
-  ): action is ActionsFromAsyncThunk<AsyncThunks[number]> => {
-    // note: this type will be correct because we have at least 1 asyncThunk
-    const matchers: [Matcher<any>, ...Matcher<any>[]] = [] as any
-
-    for (const asyncThunk of asyncThunks) {
-      matchers.push(
-        asyncThunk.pending,
-        asyncThunk.rejected,
-        asyncThunk.fulfilled,
-      )
-    }
-
-    const combinedMatcher = isAnyOf(...matchers)
-
-    return combinedMatcher(action)
+  const matchers: Matcher<any>[] = []
+
+  for (const asyncThunk of asyncThunks) {
+    matchers.push(asyncThunk.pending, asyncThunk.rejected, asyncThunk.fulfilled)
   }
+
+  return isAnyOf(...matchers)
```

Comment:
```suggestion
  return isAnyOf(...asyncThunks.flatMap(asyncThunk => [asyncThunk.pending, asyncThunk.rejected, asyncThunk.fulfilled]))
```

Reply:
the thought did occur, but i wasn't sure what the support for flatMap was like :)

Reply:
I mean, we already use it in `combineSlices` :)

Also, seems well supported: https://caniuse.com/mdn-javascript_builtins_array_flatmap

Reply:
haha, good point